fetch/data-urls/processing.any.js

```diff
@@ -0,0 +1,23 @@
+promise_test(() => fetch("resources/data-urls.json").then(res => res.json()).then(runDataURLTests), "Setup.");
+function runDataURLTests(tests) {
+  for(let i = 0; i < tests.length; i++) {
+    const input = tests[i][0];
+    let expectedMimeType = tests[i][1];
+    const expectedBody = expectedMimeType !== null ? tests[i][2] : null;
+    promise_test(t => {
+      if(expectedMimeType === "") {
+        expectedMimeType = "text/plain;charset=US-ASCII";
```

Comment:
Normalizing this in the data files would be better and clearer. Performing computation in the tests is best avoided if possible.

Reply:
Maybe, but copypasta is a risk too.

Reply:
Now I'm confused, I'm arguing in favor of keeping this as-is, isn't copypasta the least likely with the empty string as the value?

Reply:
My reply was for @domenic. I also made my comment six minutes before yours according to the GitHub timestamp on it, so not sure why it appears to be later.

Reply:
It's still really, really confusing looking at the test files and seeing empty string as the expected MIME type. It'd be quite nice if this was fixed.

If nothing else, maybe replace it with `"__default__"` or something, with a comment at the top explaining the substitution.

Reply:
Would using null work?

Reply:
null means failure, per line 6.

Reply:
The larger principle here is that any substitution requires every client of these tests to read the documentation and duplicate the substitution logic themselves, instead of just being able to write an equality test.

Reply:
Well, you have to read the documentation either way. I don't understand why the focus is on consumption so much as writing tests should be easy too. Maybe once this is implemented in more places and we have a good grasp on coverage we can simplify the consumption though.

Reply:
The focus is on being able to read the data file and tell what the test is. Code (and tests) are read more often than they are written.

If you're just going to generate tests from logic, instead of using a data file with the correct data, you might as well just use jsdom/data-url and check that the results match, instead of using a data file at all. That's obviously more extreme, but it's the same slippery slope.